apis/secrets/v1alpha1/storeconfig_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// A StoreConfigSpec defines the desired state of a ProviderConfig.
+type StoreConfigSpec struct {
+	xpv1.SecretStoreConfig `json:",inline"`
+}
+
+// A StoreConfigStatus represents the status of a StoreConfig.
+type StoreConfigStatus struct {
+	xpv1.ConditionedStatus `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// A StoreConfig configures how Crossplane controllers should store connection details.
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:printcolumn:name="TYPE",type="string",JSONPath=".spec.type"
+// +kubebuilder:printcolumn:name="DEFAULT-SCOPE",type="string",JSONPath=".spec.defaultScope"
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,store}
+// +kubebuilder:subresource:status
+type StoreConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   StoreConfigSpec   `json:"spec"`
+	Status StoreConfigStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// StoreConfigList contains a list of StoreConfig
+type StoreConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []StoreConfig `json:"items"`
+}
+
+// GetStoreConfig returns SecretStoreConfig
+func (in *StoreConfig) GetStoreConfig() xpv1.SecretStoreConfig {
+	return in.Spec.SecretStoreConfig
+}
+
+// GetCondition of this StoreConfig.
+func (in *StoreConfig) GetCondition(ct xpv1.ConditionType) xpv1.Condition {
+	return in.Status.GetCondition(ct)
+}
```

Comment:
Hrm, do we expect store configs to have conditions?

Reply:
Initially, I was thinking of some sort of connection/validation check which attempts to connect to the configured secret store and make as ready if successful. But now I would prefer to implement this later if needed/requested. 
Removed for now.